src/Tizen.NUI/src/public/Window.cs

```diff
@@ -1364,14 +1410,8 @@ public void AddFrameRenderedCallback(FrameCallbackType callback, int frameId)
         [EditorBrowsable(EditorBrowsableState.Never)]
         public void AddFramePresentedCallback(FrameCallbackType callback, int frameId)
         {
-            IntPtr ip = Marshal.GetFunctionPointerForDelegate<Delegate>(callback);
-
-            if (IntPtr.Zero == ip)
-            {
-                throw new ArgumentNullException(nameof(callback), "FrameCallbackType should not be null");
-            }
-
-            Interop.WindowInternal.AddFramePresentedCallback(swigCPtr, new HandleRef(this, ip), frameId);
+            addInterHookCallback(callback, frameId);
+            Interop.WindowInternal.AddFramePresentedCallback(swigCPtr, new HandleRef(this, Marshal.GetFunctionPointerForDelegate<Delegate>(internalHookFrameCallback)), key);
```

Comment:
Previously, "frameId" was passed as the last parameter. It's ok to change it to "key" ?
I don't know well about the usage of this parameter, but "key" seems to be existed as a counter of addInterHookCallback.

Reply:
could you check the native sample code : https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-demo/+/237623/2/examples/hello-world/hello-world-example.cpp

the frameId in AddFrameRenderedCallback is the value that user induce and it is passed and shown in user defined callback (FrameCallbackType). 

I changed it a little bit and in AddInterHookCallback() user's frameId is saved in internalHookCallbackType instance. The assignedKey is increased and is unique value so it is assigned in AddFrameRenderedCallback() to find saved user callback and frameId in the Dictionary of frameCallbackList.

It could be overhead but it is required to avoid null reference exception occured by garbage collected/unreachable delegate problem.

Reply:
Thanks for explanation. I can understand clearly :)